src/extension.ts

```diff
@@ -424,3 +423,33 @@ export async function activate(context: vscode.ExtensionContext): Promise<void>
         context.subscriptions.push(createClient(dirToAnalyze).start());
     }
 }
+
+// Search all the workspace folders and return the first that contains .phan/config.php
+// This will return the folder name in an array as required, and just only one folder,
+// the first met. We can easy modify the function to add all valid workspace folders to the array.
```

Comment:
This comment is out of date - this is already returning all valid workspace folders that **were open** when this extension started (or is unpaused?). I also see onDidChangeWorkspaceFolders exists in https://code.visualstudio.com/api/references/vscode-api#workspace , but that's out of scope.

Reply:
This is when extension is activated, so it runs with activate() function. Rewrite the comment to be:
`// Search all the workspace folders and return the ones that contains .phan/config.php file.`

The function makes exactly what the title says. Maybe in the future add a new function like "handleWorkspaceFoldersChanged()" and bound to the event 'onDidChangeWorkspaceFolders', that will call findValidProjectDirectories() and updated/restart the extension.